### ISSUE TITLE
`kubeflow-authn` (`auth-proxy`) component: support cluster local OIDC issuer

### ISSUE THREAD

**Issue by oginskis** [agilestacks/kubeflow-components#39]
In some setups, the `auth-proxy` cannot make calls to OIDC endpoints of Dex through an external load balancer (firewall, restrictive security requirements of cluster inbound traffic, etc.)

To address this issue I propose introducing a separate `dex.issuer.internal` property to point cluster local endpoint of `dex`.

The `auth-proxy` will use the internal endpoint to make direct calls to the dex, for example, to exchange an AUTH_CODE received from the user to a JWT token

The external `dex.issuer` endpoint still will be used in callbacks to make sure it opens in user browsers.

If the deployer does not provide `dex.issuer.internal` it will default to `dex.issuer` value to ensure backwards compatibility

**Comment by akranga** [comment]
Imho in this case it is ok on a stack-level overload `dex.issuer` parameter. There is no need to add additional parameter that will be used in strange cases

**Comment by oginskis** [comment]
I need both

**Comment by oginskis** [comment]
Agreed with @akranga to not introduce any extra parameters. Users can now override `oidc.issuer` to set Kubernetes DNS name if the need to
